LandsitesTools/prepare_nlp_cases.py

```diff
@@ -1 +1,282 @@
-### Coming soon.
+import argparse
+import sys
+import os
+import json
+import re
+import subprocess
+from pathlib import Path
+from interface_settings import InterfaceSettings
+import pandas as pd
+pd.set_option('display.max_colwidth', None)
+
+########################################################################################################
+######################################### Argument parser setup ########################################
+########################################################################################################
+
+# Describe command line tool
+description = "This tool creates, builds, and sets up CTSM cases " + \
+"for the NorESM LandSites Platform. It is either using the cases specified " + \
+"in the settings.txt file or interactive command line input."
+
+# Initiate the parser
+parser = argparse.ArgumentParser(description=description)
+
+## Define arguments that need to be provided
+#parser.add_argument("--version", help="show tool version", action="store_true")
+parser.add_argument("-s","--case_name_suffix", help="optional suffixes for created case folder names", 
+                    default="")
+parser.add_argument("-p","--cfg_path", help="path to configuration file", 
+                    default="../")
+parser.add_argument("-f","--cfg_file", help="configuration file to use, default is 'settings.txt'",
+                    default="settings.txt")
+parser.add_argument("-i","--interactive", help="display and prepare cases interactively", 
+                    action="store_true")
+parser.add_argument("-n","--name_output_file", help="name of output settings file (interactive only)", 
+                    default="interactive_settings.txt")
+
+args = parser.parse_args()
+
```

Comment:
Main is too long and difficult to read. Use function (save in separate files). If I understand properly:

- Read program parameters (interactively via questions or via a file)
- Create case
- Build case
The rest should be outside (not in the main).

Reply:
@huitang-earth @emiliano-gelati how should we solve this for the time being, as this will be related to the general project structure?

Reply:
Defining functions is a good idea, especially if code is to be reused. However, I think we should first discuss the general project structure.